firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/Mutation.java

```diff
@@ -93,33 +93,25 @@ public Precondition getPrecondition() {
   }
 
   /**
-   * Applies this mutation to the given MaybeDocument for the purposes of computing a new remote
-   * document. If the input document doesn't match the expected state (e.g. it is null or outdated),
-   * an `UnknownDocument` can be returned.
+   * Applies this mutation to the given Document for the purposes of computing a new remote document
+   * If the input document doesn't match the expected state (e.g. it is null or outdated), an
+   * `UnknownDocument` may be created
    *
-   * @param maybeDoc The document to mutate. The input document can be null if the client has no
-   *     knowledge of the pre-mutation state of the document.
+   * @param document The document to mutate.
    * @param mutationResult The result of applying the mutation from the backend.
-   * @return The mutated document. The returned document may be an UnknownDocument if the mutation
-   *     could not be applied to the locally cached base document.
    */
-  public abstract MaybeDocument applyToRemoteDocument(
-      @Nullable MaybeDocument maybeDoc, MutationResult mutationResult);
+  public abstract void applyToRemoteDocument(
+      MutableDocument document, MutationResult mutationResult);
 
   /**
-   * Applies this mutation to the given MaybeDocument for the purposes of computing the new local
-   * view of a document. Both the input and returned documents can be null.
+   * Applies this mutation to the given Document for the purposes of computing the new local view of
+   * a document. Both the input and returned documents can be null.
```

Comment:
"Both the input and returned documents can be null" is now somewhat incorrect since the method now returns null. Please update.